app/models/recipe.rb

```diff
@@ -0,0 +1,12 @@
+class Recipe < ApplicationRecord
+  belongs_to :user
+  belongs_to :user, foreign_key: 'user_id'
+  has_many :recipe_foods, foreign_key: 'recipe_id', dependent: :destroy
+  has_many :foods, through: :recipe_foods, dependent: :destroy
+
+  validates :name, presence: true, length: { minimum: 3, maximum: 100 }
+  validates :preparation_time, presence: true, comparison: { greater_than: 0 }
```

Comment:
- [ ] You are unable to create a recipe because you have validation for `preparation_time`, instead of `prepration_time`. Take note of the typo error you made. The code below fixes the issue.

  ```
  validates :prepration_time, presence: true, comparison: { greater_than: 0 }
  ```

Reply:
@TedLivist , Thank you very much for your detailed CR feedback. Much appreciated.

Reply:
Hi @TedLivist ,  I wanted to thank you for your help and for the feedback you gave us.